network-api/networkapi/wagtailpages/pagemodels/campaigns.py

```diff
@@ -374,15 +374,29 @@ class BanneredCampaignPage(PrimaryPage):
     tags = ClusterTaggableManager(through=BanneredCampaignTag, blank=True)
     aside = StreamField(aside_fields, blank=True)
 
+    use_intro_from_parent = models.BooleanField(
+        default=False,
+        blank=False,
+        help_text='This field will overwrite the intro field above with the intro field from this pages parent'
+    )
+
     panel_count = len(PrimaryPage.content_panels)
-    n = panel_count - 1
+    introPanel = PrimaryPage.content_panels.pop(-2)
+
+    n = panel_count - 2
```

Comment:
Can you add some context what this is doing? And what do you think about this being a method with some tests?